src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/handlers/presubmit/InternalAdaSuitabilityReviewSuitableLetterGenerator.java

```diff
@@ -0,0 +1,77 @@
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.DocumentTag;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.CaseDetails;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.field.Document;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.handlers.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.service.DocumentCreator;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.service.DocumentHandler;
+
+@Component
+public class InternalAdaSuitabilityReviewSuitableLetterGenerator implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private final DocumentCreator<AsylumCase> internalAdaSuitabilityReviewSuitableLetterGenerator;
+    private final DocumentHandler documentHandler;
+
+    public InternalAdaSuitabilityReviewSuitableLetterGenerator(
+        @Qualifier("internalAdaSuitabilitySuitable") DocumentCreator<AsylumCase> internalAdaSuitabilityReviewSuitableLetterGenerator,
+        DocumentHandler documentHandler
+    ) {
+        this.internalAdaSuitabilityReviewSuitableLetterGenerator = internalAdaSuitabilityReviewSuitableLetterGenerator;
+        this.documentHandler = documentHandler;
+    }
+
+    public boolean canHandle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+        return callback.getEvent() == Event.ADA_SUITABILITY_REVIEW
+               && callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+               && isInternalCase(asylumCase)
+               && isAcceleratedDetainedAppeal(asylumCase);
+    }
+
+    public PreSubmitCallbackResponse<AsylumCase> handle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final CaseDetails<AsylumCase> caseDetails = callback.getCaseDetails();
+        final AsylumCase asylumCase = caseDetails.getCaseData();
+
+        Document internalAdaSuitabilitySuitableLetterDocument = internalAdaSuitabilityReviewSuitableLetterGenerator.create(caseDetails);
+
+        final AdaSuitabilityReviewDecision suitabilityDecision = asylumCase.read(SUITABILITY_REVIEW_DECISION, AdaSuitabilityReviewDecision.class).orElse(null);

```

Comment:
Should we use throw an exception here instead? Otherwise (if SUITABILITY_REVIEW_DECISION is missing) the user will think a notification was sent when it wasn't.

Reply:
Now throwing an exception here